src/Form/AdherentEmailSubscriptionType.php

```diff
@@ -43,15 +65,25 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         ;
 
         $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
-            $formData = $event->getData();
-            $form = $event->getForm();
-            if (!\in_array(AdherentEmailSubscription::SUBSCRIBED_EMAILS_CITIZEN_PROJECT_CREATION, $formData['emails_subscriptions'] ?? [], true)) {
-                $form->add('citizenProjectCreationEmailSubscriptionRadius', ChoiceType::class, [
-                    'choices' => array_merge(AdherentEmailSubscription::CITIZEN_PROJECT_DISTANCE_NOTIFICATION, ['Désactivé' => Adherent::DISABLED_CITIZEN_PROJECT_EMAIL]),
+            $data = $event->getData();
+            if (empty($data['citizenProjectCreationEmailSubscriptionRadius'])
+                || Adherent::DISABLED_CITIZEN_PROJECT_EMAIL == $data['citizenProjectCreationEmailSubscriptionRadius']
+            ) {
+                $event->getForm()->add('citizenProjectCreationEmailSubscriptionRadius', ChoiceType::class, [
+                    'choices' => array_merge(CitizenProjectNotificationDistance::DISTANCES, ['Désactivé' => Adherent::DISABLED_CITIZEN_PROJECT_EMAIL]),
                 ]);
-                $formData['citizenProjectCreationEmailSubscriptionRadius'] = -1;
-                $event->setData($formData);
+                $data['citizenProjectCreationEmailSubscriptionRadius'] = Adherent::DISABLED_CITIZEN_PROJECT_EMAIL;
+                $event->setData($data);
             }
         });
     }
+
+    public function configureOptions(OptionsResolver $resolver)
```

Comment:
You should move this method above for readability (we'll change that soon in the docs).